pkg/cmd/start/main.go

```diff
@@ -102,22 +106,39 @@ func start(cmd *cobra.Command, args []string) {
 		log.Fatal(err)
 	}
 
-	if strings.EqualFold(viper.GetString("platform"), v1alpha1.FlagPlatformAutoDetect) {
-		log.Debug("Attempting to auto-detect the platform")
-		os, err := detectOpenShift(mgr.GetConfig())
-		if err != nil {
-			log.WithError(err).Info("failed to auto-detect the platform, falling back to 'kubernetes'")
+	apiList, err := availableAPIs(mgr.GetConfig())
+	if err != nil {
+		log.WithError(err).Info("Failed to determine the platform capabilities. Auto-detected properties will fallback to their default values.")
+		viper.Set("platform", v1alpha1.FlagPlatformKubernetes)
+		viper.Set("es-provision", v1alpha1.FlagProvisionElasticsearchFalse)
+	} else {
+		// detect the platform
+		if strings.EqualFold(viper.GetString("platform"), v1alpha1.FlagPlatformAutoDetect) {
+			log.Debug("Attempting to auto-detect the platform")
+			if isOpenShift(apiList) {
+				viper.Set("platform", v1alpha1.FlagPlatformOpenShift)
+			} else {
+				viper.Set("platform", v1alpha1.FlagPlatformKubernetes)
+			}
+
+			log.WithField("platform", viper.GetString("platform")).Info("Auto-detected the platform")
+		} else {
+			log.WithField("platform", viper.GetString("platform")).Debug("The 'platform' option is explicitly set")
 		}
 
-		if os {
-			viper.Set("platform", v1alpha1.FlagPlatformOpenShift)
+		// detect whether the Elasticsearch operator is available
+		if strings.EqualFold(viper.GetString("es-provision"), v1alpha1.FlagProvisionElasticsearchAuto) {
```

Comment:
This requires deploying ES operator before Jaeger operator which creates assumptions on order how operators are deployed by OLM and get to init state. Do you know if OLM is capable of resolving this?

Reply:
Technically, only the CRD has to be deployed, but you have a good point. In cases where people know that there is/will be an operator running, they can set `es-provision` to true.